### ISSUE TITLE
show methods for MersenneTwister are too verbose

### ISSUE THREAD

**Issue by ablaom** [JuliaLang/julia#32173]
```julia
julia> using Random

julia> rng = MersenneTwister()
MersenneTwister(UInt32[0x53b6e5e7, 0x7d0e7c86, 0x0ba964d1, 0xf7a1292d], 
Random.DSFMT.DSFMT_state(Int32[-1495208182, 1073561876, -197720755, 1073545455, 
2135500416, 1073484640, -1064170348, 1073166331, -884880424, 1073492359  …  866661248, 
1073642685, -615634432, 1073148329, -1013641860, 817458982, -451623403, -1961188046, 382, 
0]), [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0  …  0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], 
UInt128[0x00000000000000000000000000000000, 0x00000000000000000000000000000000,
 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000  …
  0x00000000000000000000000000000000, 0x00000000000000000000000000000000,
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000], 1002, 0)

julia> [rng, rng]
2-element Array{MersenneTwister,1}:
 MersenneTwister(UInt32[0x53b6e5e7, 0x7d0e7c86, 0x0ba964d1, 0xf7a1292d], 
Random.DSFMT.DSFMT_state(Int32[-1495208182, 1073561876, -197720755, 1073545455, 
2135500416, 1073484640, -1064170348, 1073166331, -884880424, 1073492359  …  
866661248, 1073642685, -615634432, 1073148329, -1013641860, 817458982, -451623403, 
-1961188046, 382, 0]), [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0  …  0.0, 0.0, 0.0, 0.0, 0.0, 
0.0, 0.0, 0.0, 0.0, 0.0], UInt128[0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000  …  
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000], 
1002, 0)
 MersenneTwister(UInt32[0x53b6e5e7, 0x7d0e7c86, 0x0ba964d1, 0xf7a1292d], 
Random.DSFMT.DSFMT_state(Int32[-1495208182, 1073561876, -197720755, 1073545455, 
2135500416, 1073484640, -1064170348, 1073166331, -884880424, 1073492359  …  
866661248, 1073642685, -615634432, 1073148329, -1013641860, 817458982, -451623403, 
-1961188046, 382, 0]), [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0  …  0.0, 0.0, 0.0, 0.0, 0.0, 
0.0, 0.0, 0.0, 0.0, 0.0], UInt128[0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000  …  
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 
0x00000000000000000000000000000000, 0x00000000000000000000000000000000], 
1002, 0)
```

**Comment by rfourquet** [comment]
This is fixed by #25129, which I wanted to revive as I have a little bit of time.

**Comment by rfourquet** [comment]
Fixed in #25129 : 
```
julia> rng = MersenneTwister()
MersenneTwister(0x6714e9faa4bc3f3394cd88e2a3e6e0de)

julia> rand(rng); rand(rng, Int); rng
MersenneTwister(0x6714e9faa4bc3f3394cd88e2a3e6e0de, (0, 2256, 0, 1, 1002, 1))

julia> [rng, MersenneTwister(123)]
2-element Vector{MersenneTwister}:
 MersenneTwister(0x6714e9faa4bc3f3394cd88e2a3e6e0de, (0, 2256, 0, 1, 1002, 1))
 MersenneTwister(123)
```
